src/overview/reducer.js

```diff
@@ -5,17 +5,25 @@ import * as actions from './actions'
 const defaultState = {
     searchResult: {rows: []},
     query: '',
+    isFetching:0,
 }
 
 function setQuery(state, {query}) {
     return {...state, query}
 }
 
 function setSearchResult(state, {searchResult}) {
-    return {...state, searchResult}
+	var temp = state.isFetching-1
```

Comment:
Not sure this is the best way to maintain immutability. If there is a better way do tell I will change immediately.